ast.go

```
@@ -277,16 +289,55 @@ func newAssignStmt(
 	convertFuncName string,
 	direct bool,
 	convert bool,
+	special specialAssignmentAttributes,
 ) ast.Stmt {
-	if direct && convert {
+	switch {
+	case direct && convert:
 		panic("direct and convert cannot both be set")
+	case direct && !special.IsZero():
+		panic("direct and special cannot both be set")
+	case convert && !special.IsZero():
+		panic("convert and special cannot both be set")
 	}
+
 	if convert {
 		right = &ast.CallExpr{
 			Fun:  leftType,
 			Args: []ast.Expr{right},
 		}
 	}
+
+	switch {
+	case special.ProtobufDuration && special.ProtobufDirection == DirTo:
+		// struct-to-pointer
+		return newAssignStmtUserFunc(
+			left,
+			right,
+			"ptypes.DurationProto",
+		)
+	case special.ProtobufDuration && special.ProtobufDirection == DirFrom:
+		// pointer-to-struct
+		return newAssignMultiStmtUserFunc(
+			[]ast.Expr{left, &ast.Ident{Name: "_"}},
```

We specifically ignore the error here by assigning to the blank identifier.